src/senselab/audio/tasks/forced_alignment/forced_alignment.py

```diff
@@ -0,0 +1,747 @@
+
+    Args:
+        transcript (List[SingleSegment]): The list of transcription segments.
+        model (torch.nn.Module): The alignment model.
+        align_model_metadata (Dict[str, Any]): Metadata for the alignment model.
+        audio (Audio): The audio data.
+        device (DeviceType): The device to run the model on.
+        interpolate_method (str): The method for interpolating NaNs (default: "nearest").
+        return_char_alignments (bool): Whether to return character alignments (default: False).
+        print_progress (bool): Whether to print progress (default: False).
+        combined_progress (bool): Whether to combine progress (default: False).
+
+    Returns:
+        AlignedTranscriptionResult: The aligned transcription result.
+    """
+    audio = _prepare_audio(audio)
+    max_duration = audio.waveform.shape[1] / audio.sampling_rate
+
+    model_dictionary = align_model_metadata["dictionary"]
+    model_lang = align_model_metadata["language"]
+    model_type = align_model_metadata["type"]
+
+    transcript = _preprocess_segments(
+        transcript,
+        align_model_metadata["dictionary"],
+        align_model_metadata["language"],
+        print_progress,
+        combined_progress,
+    )
+
+    aligned_segments, word_segments = _align_segments(
+        transcript=transcript,
+        model=model,
+        model_dictionary=model_dictionary,
+        model_lang=model_lang,
+        model_type=model_type,
+        audio=audio,
+        device=device,
+        max_duration=max_duration,
+        return_char_alignments=return_char_alignments,
+        interpolate_method=interpolate_method,
+    )
+    return {"segments": aligned_segments, "word_segments": word_segments}
+
+
+def align_transcriptions(
+    audios_and_transcriptions: List[Tuple[Audio, ScriptLine]], language: Language = Language(language_code="en")
+) -> List[List[ScriptLine]]:
+    """Aligns transcriptions with the given audio using a wav2vec2.0 model.
+
+    Args:
+        audios_and_transcriptions (List[Tuple[Audio, ScriptLine]]):
+            A list of tuples where each tuple contains an audio object and its corresponding transcription.
+        language (Language): The language of the audio (default is "en").
+
+    Returns:
+        List[List[ScriptLine]]: A list of lists, where each inner list contains the aligned script lines for each audio.
+    """
+    aligned_script_lines = []
+
+    # Define the language code and load model
+    device = _select_device_and_dtype()[0]  # DeviceType object
+    model_name = DEFAULT_ALIGN_MODELS_HF.get(language.language_code, "facebook/wav2vec2-base-960h")
+
+    processor = Wav2Vec2Processor.from_pretrained(model_name)
+    model = Wav2Vec2ForCTC.from_pretrained(model_name).to(device.value)
+
+    for audio, transcription in audios_and_transcriptions:
+        audio = resample_audios([audio], SAMPLE_RATE)[0]  # Resample to fit expectations of Wav2Vec2
```

Comment:
- what if the used model is not wav2vec2?
- can you please, just raise an error if the sampling rate is not the expected rate instead of doing resampling?

Reply:
For now, it will only support wav2vec2-based architectures. There is no model parameter to align_transcriptions, so a user passing the wrong model is not a possible problem. Updated to raise an error instead of doing resampling.